snc-library.sh

```diff
@@ -170,19 +170,20 @@ function create_pvs() {
 
     for pvname in $(seq -f "pv%04g" 1 ${count}); do
         if ! ${OC} get pv "${pvname}" &> /dev/null; then
-            generate_pv "${pvdir}/${pvname}" "${pvname}" | ${OC} create -f -
+            pv_template=$(generate_pv "${pvdir}/${pvname}" "${pvname}")
+            retry ${OC} create -f <(echo "${pv_template}")
```

Comment:
I think anything involving `<` `|` won't work as expected with retry, as the first try will consume the data which was produced by `echo`, and the next tries will have nothing to read from.
```
diff --git a/tools.sh b/tools.sh
index 288bd74..d22426a 100755
--- a/tools.sh
+++ b/tools.sh
@@ -91,7 +91,13 @@ fi
 function retry {
     local retries=10
     local count=0
-    until "$@"; do
+    if [ -t 0 ]; then
+        # no pipe
+        input=""
+    else
+        # pipe, save stdin
+        input="$(cat)"
+    fi
+
+    until echo "$input" | "$@"; do
         exit=$?
         wait=$((2 ** $count))
         count=$(($count + 1))
```
might help, but I expect there will be other corner cases I'm not familiar with. An intermediate function may help (one which runs `echo ${pv_template} | ${OC} create -f` so that we can then run `retry intermediate_function ${pv_template}`)

Reply:
Created https://github.com/code-ready/snc/issues/460